modules/db/src/main/scala/gem/dao/SmartGcalDao.scala

```diff
@@ -1,10 +1,45 @@
 package gem.dao
 
 import gem.config.{F2SmartGcalKey, GcalConfig, SmartGcalKey}
-import gem.enum.{GcalBaselineType, GcalLampType}
+import gem.enum._
 import doobie.imports._
+import scalaz._, Scalaz._
 
 object SmartGcalDao {
+
+  def selectF2(k: F2SmartGcalKey, t: SmartGcalType): ConnectionIO[List[Int]] = {
+    def byLamp(l: GcalLampType): ConnectionIO[List[Int]] =
+      sql"""
+        SELECT gcal_id
+          FROM smart_f2
+         WHERE lamp      = $l :: gcal_lamp_type
+           AND disperser = ${k.disperser}
+           AND filter    = ${k.filter}
+           AND fpu       = ${k.fpu}
+      """.query[Int].list
+
+    def byBaseline(b: GcalBaselineType): ConnectionIO[List[Int]] =
+      sql"""
+        SELECT gcal_id
+          FROM smart_f2
+         WHERE baseline  = $b :: gcal_baseline_type
+           AND disperser = ${k.disperser}
+           AND filter    = ${k.filter}
+           AND fpu       = ${k.fpu}
+      """.query[Int].list
+
+    t.fold(byLamp, byBaseline)
+  }
+
+  def select(k: SmartGcalKey, t: SmartGcalType): ConnectionIO[List[GcalConfig]] =
+    for {
+      ids <- k match {
+              case f2: F2SmartGcalKey => selectF2(f2, t)
+            }
+      gcs <- ids.traverseU { GcalDao.select }.map(_.flatten)
+    } yield gcs
```

Comment:
I just added composable statement fragments to doobie so we may be able to remove the repetition.